--- tensorflow_federated/cc/core/impl/executors/BUILD
+++ tensorflow_federated/cc/core/impl/executors/BUILD
@@ -25,6 +25,7 @@ cc_library(
     name = "cardinalities",
     srcs = ["cardinalities.cc"],
     hdrs = ["cardinalities.h"],
+    visibility = ["//visibility:public"],
     deps = [
         "@com_google_absl//absl/container:btree",
         "@com_google_absl//absl/status:statusor",
@@ -362,6 +363,7 @@ cc_library(
     name = "executor",
     srcs = ["executor.cc"],
     hdrs = ["executor.h"],
+    visibility = ["//visibility:public"],
     deps = [
         ":status_macros",
         "//tensorflow_federated/proto/v0:computation_cc_proto",
@@ -498,6 +500,7 @@ cc_library(
     name = "federating_executor",
     srcs = ["federating_executor.cc"],
     hdrs = ["federating_executor.h"],
+    visibility = ["//visibility:public"],
     deps = [
         ":cardinalities",
         ":executor",
@@ -631,6 +634,7 @@ cc_library(
     name = "reference_resolving_executor",
     srcs = ["reference_resolving_executor.cc"],
     hdrs = ["reference_resolving_executor.h"],
+    visibility = ["//visibility:public"],
     deps = [
         ":executor",
         ":status_macros",
@@ -924,6 +928,7 @@ cc_library(
     name = "tensorflow_executor",
     srcs = ["tensorflow_executor.cc"],
     hdrs = ["tensorflow_executor.h"],
+    visibility = ["//visibility:public"],
     deps = [
         ":dataset_from_tensor_structures",
         ":executor",
