--- third_party/xla/xla/ffi/attribute_map.cc	2026-05-20 10:49:47.114883837 -0700
+++ third_party/xla/xla/ffi/attribute_map.cc	2026-05-20 10:50:08.789444210 -0700
@@ -285,7 +285,7 @@
         } else if constexpr (std::is_same_v<U, AttributesDictionary>) {
           *proto.mutable_dict() = value.ToProto();
         } else {
-          static_assert(false, "Unsupported attribute type");
+          static_assert(!sizeof(U*), "Unsupported attribute type");
         }
       },
       AsVariant());
@@ -320,7 +320,7 @@
         } else if constexpr (std::is_same_v<U, double>) {
           proto.set_f64(value);
         } else {
-          static_assert(false, "Unsupported scalar type");
+          static_assert(!sizeof(U*), "Unsupported scalar type");
         }
       },
       AsVariant());
@@ -408,7 +408,7 @@
           proto.mutable_f64()->mutable_values()->Assign(value.begin(),
                                                         value.end());
         } else {
-          static_assert(false, "Unsupported array type");
+          static_assert(!sizeof(U*), "Unsupported array type");
         }
       },
       AsVariant());
@@ -480,7 +480,7 @@
         } else if constexpr (std::is_same_v<U, std::string>) {
           proto.set_str(value);
         } else {
-          static_assert(false, "Unsupported flat attribute type");
+          static_assert(!sizeof(U*), "Unsupported flat attribute type");
         }
       },
       AsVariant());
