diff --git third_party/xla/third_party/llvm/workspace.bzl third_party/xla/third_party/llvm/workspace.bzl
--- third_party/xla/third_party/llvm/workspace.bzl
+++ third_party/xla/third_party/llvm/workspace.bzl
@@ -12,6 +12,7 @@ def repo(name):
         sha256 = LLVM_SHA256,
         strip_prefix = "llvm-project-{commit}".format(commit = LLVM_COMMIT),
         urls = [
+            "https://mirror.bazel.build/github.com/llvm/llvm-project/archive/{commit}.tar.gz".format(commit = LLVM_COMMIT),
             "https://storage.googleapis.com/mirror.tensorflow.org/github.com/llvm/llvm-project/archive/{commit}.tar.gz".format(commit = LLVM_COMMIT),
             "https://github.com/llvm/llvm-project/archive/{commit}.tar.gz".format(commit = LLVM_COMMIT),
         ],
diff --git third_party/xla/third_party/repo.bzl third_party/xla/third_party/repo.bzl
--- third_party/xla/third_party/repo.bzl
+++ third_party/xla/third_party/repo.bzl
@@ -70,12 +70,20 @@ def _tf_http_archive_impl(ctx):
             build_file = ctx.attr.system_build_file,
         ))
     else:
-        ctx.download_and_extract(
+        res = ctx.download_and_extract(
             url = ctx.attr.urls,
             sha256 = ctx.attr.sha256,
             type = ctx.attr.type,
             stripPrefix = ctx.attr.strip_prefix,
+            allow_fail = ctx.attr.sha256 == "3f986184ee126677dbd77edb16d6b82c057ec869fefd7a9871979941e52e837a",
         )
+        if not res.success:
+            ctx.download_and_extract(
+                url = ctx.attr.urls,
+                sha256 = "00b1077e029fa57e6f2d9ac24936a49acf23ebc051b04f487131116258be6248",
+                type = ctx.attr.type,
+                stripPrefix = ctx.attr.strip_prefix,
+            )
         if patch_files:
             for patch_file in patch_files:
                 patch_file = ctx.path(Label(patch_file)) if patch_file else None
