# Update visibility for executors to public.

diff --git tensorflow_federated/python/core/impl/executors/BUILD tensorflow_federated/python/core/impl/executors/BUILD
--- tensorflow_federated/python/core/impl/executors/BUILD
+++ tensorflow_federated/python/core/impl/executors/BUILD
@@ -56,3 +56,4 @@
 py_library(
     name = "cpp_to_python_executor",
+    visibility = ["//visibility:public"],
     srcs = ["cpp_to_python_executor.py"],

diff --git tensorflow_federated/cc/core/impl/executors/BUILD tensorflow_federated/cc/core/impl/executors/BUILD
--- tensorflow_federated/cc/core/impl/executors/BUILD
+++ tensorflow_federated/cc/core/impl/executors/BUILD
@@ -306,6 +306,7 @@
 cc_library(
     name = "executor_service",
     srcs = ["executor_service.cc"],
     hdrs = ["executor_service.h"],
+    visibility = ["//visibility:public"],
     deps = [
         ":cardinalities",
         ":executor",
