# Enable the regex feature for oak_attestation_verification.
--- oak_attestation_verification/BUILD
+++ oak_attestation_verification/BUILD
@@ -41,11 +41,13 @@
 rust_library(
     name = "oak_attestation_verification",
     srcs = glob(
         ["src/**"],
         exclude = RESULT_SRC,
     ),
     compile_data = [
         "//oak_attestation_verification/data:amd_ask_certs",
         "//oak_attestation_verification/data:intel_pck_root",
     ],
+    crate_features = ["regex"],
     deps = [
+        "@oak_crates_index//:regex-lite",
