diff --git tensorflow_federated/cc/core/impl/aggregation/core/BUILD tensorflow_federated/cc/core/impl/aggregation/core/BUILD
index 6f681968a..397e35695 100644
--- tensorflow_federated/cc/core/impl/aggregation/core/BUILD
+++ tensorflow_federated/cc/core/impl/aggregation/core/BUILD
@@ -16,7 +16,7 @@ load("@com_github_grpc_grpc//bazel:python_rules.bzl", "py_proto_library")
 load("@rules_cc//cc:cc_binary.bzl", "cc_binary")
 load("@rules_cc//cc:cc_library.bzl", "cc_library")
 load("@rules_cc//cc:cc_test.bzl", "cc_test")
-load("@rules_cc//cc:defs.bzl", "cc_proto_library")
+load("@com_google_protobuf//bazel:cc_proto_library.bzl", "cc_proto_library")
 
 package(
     default_applicable_licenses = ["//:package_license"],
