diff --git fcp/confidentialcompute/python/BUILD fcp/confidentialcompute/python/BUILD
index 2855692..a7df79c 100644
--- fcp/confidentialcompute/python/BUILD
+++ fcp/confidentialcompute/python/BUILD
@@ -33,9 +33,9 @@ py_library(
     visibility = ["//visibility:public"],
     deps = [
         "@federated_language//federated_language",
+        "@jax_privacy//jax_privacy",
         "@org_tensorflow_federated//tensorflow_federated",
         "@pypi//jax",
-        "@pypi//jax_privacy",
         "@pypi//optax",
         "@pypi//tensorflow",
     ],