--- oak_remote_attestation/proto/v1/BUILD
+++ oak_remote_attestation/proto/v1/BUILD
@@ -16,9 +16,7 @@

 load("@bazel_skylib//rules:build_test.bzl", "build_test")
 load("@com_github_grpc_grpc//bazel:cc_grpc_library.bzl", "cc_grpc_library")
-load("@io_grpc_grpc_java//:java_grpc_library.bzl", "java_grpc_library")
 load("@rules_cc//cc:defs.bzl", "cc_proto_library")
-load("@rules_java//java:defs.bzl", "java_proto_library")
 load("@rules_proto//proto:defs.bzl", "proto_library")

 package(
@@ -37,11 +35,6 @@ cc_proto_library(
     deps = [":messages_proto"],
 )

-java_proto_library(
-    name = "messages_java_proto",
-    deps = [":messages_proto"],
-)
-
 proto_library(
     name = "service_streaming_proto",
     srcs = ["service_streaming.proto"],
@@ -78,35 +71,11 @@ cc_grpc_library(
     deps = [":service_unary_cc_proto"],
 )

-java_proto_library(
-    name = "service_streaming_java_proto",
-    deps = [":service_streaming_proto"],
-)
-
-java_grpc_library(
-    name = "service_streaming_java_grpc",
-    srcs = [":service_streaming_proto"],
-    deps = [":service_streaming_java_proto"],
-)
-
-java_proto_library(
-    name = "service_unary_java_proto",
-    deps = [":service_unary_proto"],
-)
-
-java_grpc_library(
-    name = "service_unary_java_grpc",
-    srcs = [":service_unary_proto"],
-    deps = [":service_unary_java_proto"],
-)
-
 build_test(
     name = "build_test",
     targets = [
         ":messages_proto",
         ":service_streaming_cc_grpc",
-        ":service_streaming_java_grpc",
         ":service_unary_cc_grpc",
-        ":service_unary_java_grpc",
     ],
 )
