# Add extra link flags when building restricted kernel binaries.
--- bazel/rust/defs.bzl
+++ bazel/rust/defs.bzl
@@ -46,2 +46,8 @@
         extra_rustc_flags = {
-            "x86_64-unknown-none": ["-C", "target-feature=+avx,+avx2,-soft-float"],
+            "x86_64-unknown-none": [
+                "-Crelocation-model=static",
+                "-Ccode-model=small",
+                "-Ctarget-feature=+sse,+sse2,+ssse3,+sse4.1,+sse4.2,+avx,+avx2,+rdrand,-soft-float",
+                "-Ctarget-cpu=x86-64-v3",
+                "-Clink-arg=-zmax-page-size=0x200000",
+            ],
