diff --git tensorflow_federated/cc/core/impl/executors/BUILD tensorflow_federated/cc/core/impl/executors/BUILD
index ec09dc818..41e1dc22f 100644
--- tensorflow_federated/cc/core/impl/executors/BUILD
+++ tensorflow_federated/cc/core/impl/executors/BUILD
@@ -45,7 +45,7 @@ cc_library(
         "@com_google_absl//absl/status:statusor",
         "@com_google_absl//absl/strings",
         "@com_google_absl//absl/strings:string_view",
-        "@eigen",
+        "@eigen_archive//:eigen3",
         "@federated_language//federated_language/proto:array_cc_proto",
         "@federated_language//federated_language/proto:data_type_cc_proto",
     ],
@@ -1034,7 +1034,7 @@ cc_library(
         "@com_google_absl//absl/strings",
         "@com_google_absl//absl/strings:cord",
         "@com_google_absl//absl/types:span",
-        "@eigen",
+        "@eigen_archive//:eigen3",
         "@federated_language//federated_language/proto:array_cc_proto",
         "@federated_language//federated_language/proto:data_type_cc_proto",
         "@org_tensorflow//tensorflow/core:framework",
@@ -1057,7 +1057,7 @@ cc_test(
         "@com_google_absl//absl/status",
         "@com_google_absl//absl/status:statusor",
         "@com_google_absl//absl/strings:string_view",
-        "@eigen",
+        "@eigen_archive//:eigen3",
         "@federated_language//federated_language/proto:array_cc_proto",
         "@federated_language//federated_language/proto:computation_cc_proto",
         "@org_tensorflow//tensorflow/core:framework",