diff --git program_executor_tee/program_context/cc/BUILD program_executor_tee/program_context/cc/BUILD
--- program_executor_tee/program_context/cc/BUILD
+++ program_executor_tee/program_context/cc/BUILD
@@ -138,3 +138,4 @@
         "@federated-compute//fcp/base",
         "@federated-compute//fcp/confidentialcompute:cose",
+        "@federated-compute//fcp/confidentialcompute:crypto",
         "@federated-compute//fcp/protos/confidentialcompute:blob_header_cc_proto",
@@ -330,3 +331,4 @@
         "@federated-compute//fcp/protos/confidentialcompute:program_worker_cc_proto",
+        "@com_github_grpc_grpc//:grpc++",
         "@oak//cc/oak_session:config",
         "@oak//cc/oak_session:server_session",
diff --git program_executor_tee/BUILD program_executor_tee/BUILD
--- program_executor_tee/BUILD
+++ program_executor_tee/BUILD
@@ -102,10 +102,12 @@
         "@com_google_absl//absl/log",
+        "@com_google_absl//absl/base:core_headers",
         "@com_google_absl//absl/log:check",
         "@com_google_absl//absl/log:die_if_null",
         "@com_google_absl//absl/status",
         "@com_google_absl//absl/status:statusor",
         "@com_google_absl//absl/strings",
         "@com_google_absl//absl/strings:str_format",
         "@com_google_absl//absl/synchronization",
         "@com_google_protobuf//:protobuf",
+        "@com_google_protobuf//:protobuf_lite",
         "@confidential-federated-compute//containers:blob_metadata",
