--- containers/BUILD
+++ containers/BUILD
@@ -180,5 +180,4 @@
         "@federated-compute//fcp/protos/confidentialcompute:confidential_transform_cc_grpc",
         "@federated-compute//fcp/protos/confidentialcompute:confidential_transform_cc_proto",
-        "@federated-compute//fcp/protos/confidentialcompute:fed_sql_container_config_cc_proto",
         "@federated-compute//fcp/protos/confidentialcompute:kms_cc_proto",
         "@oak//cc/crypto:encryption_key",
